### PR TITLE
Support WW detection via GPIO pin INT_N 

### PR DESCRIPTION
**Pull request by lucianomartin**
This feature is enabled by using the CLI flag '-g'.

The SDK reads the INT_N pin and it will indicate a WW detection when the pin goes from high to low.

### CODE REVIEW

File: KWD/GPIO/src/GPIOKeywordDetector.cpp [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    std::chrono::milliseconds msToPushPerIteration) :
+        AbstractKeywordDetector(keyWordObservers, keyWordDetectorStateObservers),
+        m_stream{stream},
+        m_maxSamplesPerPush((audioFormat.sampleRateHz / HERTZ_PER_KILOHERTZ) * msToPushPerIteration.count()) {
+}
+
+GPIOKeywordDetector::~GPIOKeywordDetector() {
+    m_isShuttingDown = true;
+    if (m_detectionThread.joinable())
+        m_detectionThread.join();
```
Comment (ed-xmos) [inline KWD/GPIO/src/GPIOKeywordDetector.cpp:166]: What happens if not joinable()?

File: KWD/GPIO/src/GPIOKeywordDetector.cpp [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+            {
+                ACSDK_INFO(LX("WW detected"));
+                notifyKeyWordObservers(
+                    m_stream,
+                    WAKEWORD_STRING,
+                    // avsCommon::sdkInterfaces::KeyWordObserverInterface::UNSPECIFIED_INDEX,
+                    (m_streamReader->tell() < (m_maxSamplesPerPush*KW_REWIND_SAMPLES) ? 0 : m_streamReader->tell() - (m_maxSamplesPerPush*KW_REWIND_SAMPLES)),
+                    m_streamReader->tell());
+            }
+            oldGpioValue = gpioValue;
```
Comment (ed-xmos) [inline KWD/GPIO/src/GPIOKeywordDetector.cpp:231]: Cool - edge detection

File: tools/Install/setup.sh [hunk 3]
```diff
@@ -123,7 +123,8 @@ show_help() {
   echo  '  -a <file-name>        The file that contains Android installation configurations (e.g. androidConfig.txt)'
   echo  '  -d <description>      The description of the device.'
   echo  '  -m <manufacturer>     The device manufacturer name.'
-  echo  '  -x <xmos-device-type> XMOS device to setup: default xvf3510, possible value xvf3500'
+  echo  '  -x <xmos-device-type> XMOS device to setup: default xvf3510'
```
Comment (ed-xmos) [inline tools/Install/setup.sh:126]: worth adding other possible values? Old comment was handy

**Review by ed-xmos — APPROVED** [review]
A lot of work here. Nice C++ skills... 
Only minor comments. Your call to address (and I will rubber stamp) or merge as is.

File: KWD/GPIO/src/GPIOKeywordDetector.cpp (hunk 1 above)
Comment (lucianomartin) [reply KWD/GPIO/src/GPIOKeywordDetector.cpp:166]: This is a pretty standard operation in the AVS, I don't know the details to answer your question I am afraid. Do you have any suggestion?

**Pull request merged**